subsys/net/ip/tcp.c

```diff
@@ -580,6 +593,10 @@ static int tcp_conn_unref(struct tcp *conn)
 	(void)k_work_cancel_delayable(&conn->fin_timer);
 	(void)k_work_cancel_delayable(&conn->persist_timer);
 	(void)k_work_cancel_delayable(&conn->ack_timer);
+	(void)k_work_cancel_delayable(&conn->send_timer);
```

Comment:
Just wondering, why not move the canceling of the work items further up to before the k_mutex_unlock(&conn->lock); in line 539 to prevent it from another time acquiring the lock, which may cause issues with potential deadlocks?

Reply:
Well, there's a ref-count check on line 542 that bails out early, so we shouldn't cancel unless that cnt == 0